agent/engine/execcmd/manager_start_linux.go

```diff
@@ -103,6 +107,10 @@ func (m *manager) StartAgent(ctx context.Context, client dockerapi.DockerClient,
 		return nil
 	}
 	ma, _ := container.GetManagedAgentByName(ExecuteCommandAgentName)
+	if HasExecAgentInitFailed(ma) {
```

Comment:
hmm, sounds to me that we should not call this method if init failed. This is going to be confusing in the logs.